compiler/fm/src/lib.rs

```diff
@@ -111,14 +109,19 @@ impl FileManager {
             anchor_path.join(format!("{mod_name}.{FILE_EXTENSION}"))
         };
 
-        self.add_file(&candidate).ok_or_else(|| candidate.as_os_str().to_string_lossy().to_string())
+        self.name_to_id(candidate.clone())
+            .ok_or_else(|| candidate.as_os_str().to_string_lossy().to_string())
     }
 
+    // TODO: This should accept a &Path instead of a PathBuf
```

Comment:
Not sure it should as it will just immediately create a pathbuf again. We should probably rename `from_path` to `from_path_buf` though.

Reply:
I think that's an issue with PathString that has leaked into `name_to_id` -- since I shouldn't need to allocate in order to use something as a key lookup. We could probably get rid of PathString and or just use a type alias, though I was going to leave that to another PR